### DISCUSSION TITLE
`vitest run` always exits with code 0

### DISCUSSION THREAD

**Discussion by ivkremer** [vitest-dev/vitest#5249] [Q&A]
Hi,

I have the following script in my `package.json`:

```json
"coverage": "vitest run --coverage --silent"
```

When I run it or also run `vitest run -u` I have zero exit code regardless of the fact whether there were failed tests or the coverage was below the declared thresholds.

I'm using vitest version `0.28.4`. I've noticed that it works the same in `0.34.6` but changes somewhere in version `1.*`. Though I don't want to update the library right now.

Is there a way to have different exit codes depending on the result of the command execution? Also can it be considered a bug or I just misunderstand the concept and missing something here?

**Comment by kris-sum** [comment]
Did you also update the matching package `@vitest/coverage-istanbul` (or v8) to 1.* to match your vitest version?

**Reply by ivkremer** [reply]
Sorry, I didn't try so far but can I update istanbul to 1.* but keep the main library as 0.28.4?
The thing is this behavior is getting fixed when updating `vitest` to `1.*`. It exits with e.g. code `1` when a test or a coverage check was failed. But with version `0.28.4` being installed it exits with `0` always.

**Reply by ivkremer** [reply]
Sorry, just came back to this problem. It's not `@vitest/coverage-isnatbul` in our case but `"@vitest/coverage-c8": "^0.28.4"`. So the version is matching but it's c8. Though `npm run test` always exists with `0` regardless of the outcome.
